test/translator/report_luns_test.cc

```diff
@@ -0,0 +1,99 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "lib/translator/report_luns.h"
+#include "lib/translator/common.h"
+
+#include "absl/base/casts.h"
+#include "gtest/gtest.h"
+
+// Report Luns Translation Tests
+
+namespace {
+
+TEST(reportLunsToNvme, shouldReturnCorrectCommand) {
+  nvme_defs::GenericQueueEntryCmd nvme_cmd = {};
+  scsi_defs::ReportLunsCommand scsi_cmd = {};
+  uint64_t expected_prp = 2323;
+  uint32_t expected_alloc_len = 344;
+  scsi_cmd.alloc_length = expected_alloc_len;
+
+  uint8_t* buf_ptr = reinterpret_cast<uint8_t*>(&scsi_cmd);
+  absl::Span<uint8_t> scsi_cmd_span =
+      absl::MakeSpan(buf_ptr, sizeof(scsi_defs::ReportLunsCommand));
+
+  auto alloc_callback = [](uint16_t count) -> void* {
+    EXPECT_EQ(1, count);
+    return reinterpret_cast<void*>(2323);
+  };
+  void (*dealloc_callback)(void*, uint16_t) = nullptr;
+  translator::SetAllocPageCallbacks(alloc_callback, dealloc_callback);
+
+  uint32_t actual_alloc_len;
+  translator::StatusCode actual_status =
+      translator::ReportLunsToNvme(scsi_cmd_span, nvme_cmd, actual_alloc_len);
+
+  EXPECT_EQ(translator::StatusCode::kSuccess, actual_status);
+  EXPECT_EQ(static_cast<uint8_t>(nvme_defs::AdminOpcode::kIdentify),
+            nvme_cmd.opc);
+  EXPECT_EQ(expected_prp, nvme_cmd.dptr.prp.prp1);
+  EXPECT_EQ(0x2, nvme_cmd.cdw[0]);
+  EXPECT_EQ(expected_alloc_len, actual_alloc_len);
+}
+
+TEST(reportLunsToScsi, shouldFillBufferCorrectly) {
+  nvme_defs::IdentifyNamespaceList ns_list;
+  uint32_t ns_list_size = 125;
+  for (uint32_t i = 0; i < ns_list_size; ++i) {
+    ns_list.ids[i] = i + 1;
+  }
+  nvme_defs::GenericQueueEntryCmd identify_cmd = {};
+  identify_cmd.dptr.prp.prp1 = reinterpret_cast<uint64_t>(&ns_list);
+
+  uint32_t lun_list_byte_size = sizeof(scsi_defs::LunAddress) * ns_list_size;
+  uint32_t buf_len =
+      sizeof(scsi_defs::ReportLunsParamData) + lun_list_byte_size;
+  uint8_t buffer[buf_len];
+  absl::Span<uint8_t> span = absl::MakeSpan(buffer, buf_len);
+
+  translator::StatusCode actual_status =
+      translator::ReportLunsToScsi(identify_cmd, span);
+
+  // Verify status code & byte length field
+  EXPECT_EQ(translator::StatusCode::kSuccess, actual_status);
+  scsi_defs::ReportLunsParamData actual_response;
+  translator::ReadValue(span, actual_response);
+  EXPECT_EQ(lun_list_byte_size, actual_response.list_byte_length);
+
+  // Verify list contents
+  scsi_defs::LunAddress* lun_list;
+  lun_list = reinterpret_cast<scsi_defs::LunAddress*>(buffer +
+                                                      sizeof(actual_response));
+  for (scsi_defs::LunAddress i = 0; i < ns_list_size; ++i) {
+    EXPECT_EQ(i, lun_list[i]);
+  }
+}
+
+TEST(reportLunsToScsi, shouldFailNotEnoughMemory) {
+  nvme_defs::GenericQueueEntryCmd cmd_data;
+  uint8_t* buf = nullptr;
+  absl::Span<uint8_t> span = absl::MakeSpan(buf, 0);
+
+  translator::StatusCode actualStatus =
```

Comment:
nit: actual_status

Reply:
Nice catch! So hard to let go of my camelcase ways lol